lib/routes/cspengyuan/index.ts

```diff
@@ -0,0 +1,309 @@
+import { Route } from '@/types';
+import { load } from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+import timezone from '@/utils/timezone';
+import cache from '@/utils/cache';
+import logger from '@/utils/logger';
+import puppeteer from '@/utils/puppeteer';
+import got from 'got';
+import https from 'https';
+
+export const route: Route = {
+    path: '/credit-research/:category/:type?',
+    name: '信用研究',
+    url: 'www.cspengyuan.com/pengyuancmscn/',
+    maintainers: ['orzchen'],
+    example: '/cspengyuan/credit-research/macro',
+    parameters: {
+        category: '（必须）匹配一级分类，例如 macro、bond-market、industry 等。',
+        type: '（可选）匹配报告类型或细节类型，例如 new、weekly、monthly、subject 等。',
+        page: '（可选）匹配页数，例如 1,4、1,-1、-4,-1 等。',
+    },
+    description: `::: TIP
+        **base route**: \`/cspengyuan/\`
+
+        默认情况下只获取第一页的最新数据，通过添加查询参数 page 来定义查询的页面数量，如 page=1,4 则为第1页至第4页，为负数则为反向索引。
+
+        但是由于缓存的原因这里使用参数查询无法及时更新缓存。
+
+        过滤了 文章/PDF 链接为空的文章。
+
+        通过添加查询参数 image 来获取 PDF 第一页截图。
+
+        同时使用 page 和 image 参数可能会导致触发反爬并严重影响服务器开销。
+
+        |       宏观研究        |            结构融资研究            |        评级研究        |       国际研究       |
+        | :-------------------: | :--------------------------------: | :--------------------: | :------------------: |
+        | credit-research/macro | credit-research/structured-finance | credit-research/rating | credit-research/intl |
+
+        | **债市研究** |             专题研究             |                热点分析                 |              债市周报              |              债市月报               |              债市年报              |
+        | :----------: | :------------------------------: | :-------------------------------------: | :--------------------------------: | :---------------------------------: | :--------------------------------: |
+        |      ×       | credit-research/industry/comment | credit-research/bond-market/hot-comment | credit-research/bond-market/weekly | credit-research/bond-market/monthly | credit-research/bond-market/annual |
+
+        | **行业研究** |             行业点评             |           行业信用展望           |             行业专题             |
+        | :------: | :------------------------------: | :------------------------------: | :------------------------------: |
+        |    ×     | credit-research/industry/comment | credit-research/industry/outlook | credit-research/industry/subject |
+
+        | **出版物** |                  期刊                  |                 专著                  |
+        | :----: | :------------------------------------: | :-----------------------------------: |
+        |   ×    | credit-research/publication/periodical | credit-research/publication/monograph |
+    :::`,
```

Comment:
Indent the content with either 0 or 2 spaces. Otherwise the markdown content won't be rendered properly.